pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSchemaRestletResource.java

```diff
@@ -207,7 +219,18 @@ public Representation put(Representation entity) {
   private Representation uploadSchema(
       @Parameter(name = "schemaName", in = "path", description = "The name of the schema to get")
       String schemaName) throws Exception {
-    File dataFile = getUploadContents();
+    File dataFile;
+    try {
+      dataFile = getUploadContents();
+    } catch (FileUploadBase.InvalidContentTypeException e) {
+      LOGGER.info("Invalid content type while updating schema {}", schemaName);
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST,
+          e.getMessage());
+    } catch (Exception e) {
+      LOGGER.error("Error reading request body while updating new schema: {}", schemaName, e);
```

Comment:
Making separate method will require different exception to allow early return. It also needs if-else to handle optional schemaName in logs. So, there is no real benefit.

We can make a common method that handles schema update that is called by both PUT and POST apis. But there are subtle differences and no UTs. So, I want to limit the scope of this change.